src/drone.py

```diff
@@ -0,0 +1,172 @@
+import roslibpy
+from abc import ABC, abstractmethod
+from enum import Enum
+
+class Drone(ABC):
+
+    class Flight_Status(Enum):
+        ON_GROUND_STANDBY = 1
+        IN_AIR_STANDBY = 2
+        FLYING = 3
+        FLYING_HOME = 4
+        PAUSED_IN_AIR = 5
+        LANDING = 6
+        NULL = 7
+
+    class UpdateMissionAction(Enum):
+        CONTINUE_MISSION = 0,
+        UPDATE_CURRENT_MISSION = 1,
+        END_AND_HOVER = 2
+
+    class WaypointActions(Enum):
+        START = 0
+        STOP = 1
+        PAUSE = 2
+        RESUME = 3
+
+    # Speed of drone in flight
+    speed = 5
+    
+    def __init__(self, drone_name, drone_type, id=None):
+        self.id = id
+        self.drone_type = drone_type
+        self.drone_name = drone_name
+        self.connection_status = False
+        self.flight_status = Drone.Flight_Status.NULL
+        self.topics = []
+        self.services = []
+        self.sensors = []
+        self.mission_msg_list = []
+        self.waypoints = []
+        self.waypoints_count = 0
+        #TODO: define position structure i.e. (lat, long) or (x, y);
+        # tuple vs dictionary
+        self.position = None
+        from operator import ROS_master_connection as server_connection
+        self.ROS_master_connection = server_connection
+
+
+    @staticmethod
+    def create(drone_name, drone_type, id=None):
+        from djimatrice_drone import DjiMatriceDrone
+        from mavros_drone import MavrosDrone
+        drones = {
+            "DjiMatrice": DjiMatriceDrone, 
```

Comment:
I think we should try to make these keys consistent, so we should use either "DjiMatriceDrone" and "MavrosDrone" or "DjiMatrice" and "Mavros".

Reply:
Lets go with "DjiMatrice" and "Mavros"